drivers/tty/serdev/pibridge.c

```diff
@@ -400,34 +402,45 @@ int pibridge_req_io(u8 addr, u8 cmd, u8 *snd_buf, u16 snd_len, u8 *rcv_buf,
 			"receive head error in io-req\n");
 		return -EIO;
 	}
+	crc = pibridge_crc8(0, (u8 *) &pkthdr, sizeof(pkthdr));
+	to_receive = min((u16) pkthdr.len, rcv_len);
+	to_discard = pkthdr.len - to_receive;
 
-	if (rcv_len != 0) {
-		if (pibridge_recv(rcv_buf, rcv_len) != rcv_len) {
+	if (to_receive) {
+		if (pibridge_recv(rcv_buf, to_receive) != to_receive) {
 			dev_warn_ratelimited(&pibridge_s->serdev->dev,
-				"receive data error in io-req\n");
-			return -EBADMSG;
+				"receive data error in gate-req(len:%d)\n",
+				to_receive);
+			return -EIO;
 		}
 	}
 
-	if (pibridge_recv(&crc_rcv, sizeof(u8)) != sizeof(u8)) {
-		dev_warn_ratelimited(&pibridge_s->serdev->dev,
-			"receive crc error in io-req\n");
-		return -EIO;
-	}
-
-	crc = pibridge_crc8(0, (u8 *) &pkthdr, sizeof(pkthdr));
-	crc = pibridge_crc8(crc, rcv_buf, rcv_len);
-
-	if (crc != crc_rcv) {
+	if (to_discard) {
+		/* The provided buffer was too small. Discard the rest of the
+		   received data as well as the following CRC checksum byte. */
+		if (pibridge_discard_timeout(to_discard + 1,
+					     PIBRIDGE_IO_TIMEOUT))
+			dev_warn_ratelimited(&pibridge_s->serdev->dev,
+				"failed to discard %u bytes within timeout\n",
+				to_discard);
 		dev_warn_ratelimited(&pibridge_s->serdev->dev,
-			"check crc error in io-req\n");
+			"received packet truncated (%u bytes missing)\n",
+			to_discard);
 		return -EBADMSG;
+	} else {
```

Comment:
You're unconditionally calling `return` in the `if` branch. So you don't need an `else` branch.

Drop the `else` branch and move everything inside it 1 tab to the left.

Reply:
Same issue exists in `pibridge_req_gate_tmt()`.

Reply:
Agreed, this is a leftover from an earlier implementation, I will remove the superfluous else branch.

Reply:
Thanks. Would you be willing to amend `pibridge_req_gate_tmt()` in the same way or do you prefer to do that in a separate pull? If the latter, creation of a task would probably be necessary lest we forget it.